libs/utils/test.py

```diff
@@ -100,4 +102,24 @@ def _experimentsFinalize(cls):
         Code executed after running the experiments
         """
 
+@wrapt.decorator
+def experiment_test(wrapped_test, instance, args, kwargs):
+    """
+    Convert a LisaTest test method to be automatically called for each experiment
+
+    The method will be passed the experiment object and a list of the names of
+    tasks that were run as the experiment's workload.
+    """
+    for experiment in instance.executor.experiments:
+        tasks = experiment.wload.tasks.keys()
+        try:
+            wrapped_test(experiment, tasks, *args, **kwargs)
+        except AssertionError as e:
+            trace_relpath = os.path.join(experiment.out_dir, "trace.dat")
+            e.args += ("Check trace file: " + os.path.abspath(trace_relpath),)
```

Comment:
See my previous message regarding capturing logging.  If you see the output it gets really messy: you first see the normal logging and then when there is a failure, you see the all the logging again with the traceback.  If you have multiple failures, it is very hard to see them all.  

My personal preference is what this patch does and run nose with `--nologcapture`.